app/backend/pyproject.toml

```diff
@@ -1,12 +1,29 @@
-[tool.black]
+[tool.ruff]
 line-length = 120
 
-[tool.isort]
-skip_gitignore = true
-include_trailing_comma = true
-multi_line_output = 3
-line_length = 120
-known_first_party = ["couchers", "proto"]
+[tool.ruff.lint]
+select = [
+    "E",   # pycodestyle
+    "F",   # pyflakes
+    "UP",  # pyupgrade
+    "B",   # flake8-bugbear
+    "C4",  # flake8-comprehensions
+    "A",   # flake8-builtins
+    "I",   # isort
+]
+ignore = [
+    "B007",   # Loop control variable not used within loop body
+    "E501",   # Line too long
+    "E711",   # Comparison to `None` should be `cond is None`
+    "E712",   # Avoid equality comparisons to `True`; use `if approved:` for truth checks
+    "F811",   # Redefinition of unused variable
+    "F841",   # Local variable is assigned to but never used
+    "UP015",  # Unnecessary open mode parameters
+]
+
+[tool.ruff.lint.isort]
+known-first-party = ["couchers", "proto", "tests", "dummy_data"]
+extra-standard-library = ["zoneinfo"]
```

Comment:
Hmm, this should be in 3.12?

Reply:
Yes, it is. But I have noticed that the ruff isort module is not always up-to-date with these, so you sometimes need to declare a standard-library like this.